tests/testthat/custom_tests/test-regional-dataset.R

```diff
@@ -1,135 +1,24 @@
-    }
-  )
-}
-
 test_regional_dataset <- function(source, level, download = FALSE) {
-  data_name <- paste0(source, " at level ", level)
   region <- eval(parse(
     text = paste0(
       source, "$new(level = '", level,
       "', verbose = FALSE)"
     )
   ))
+  snapshot_path <- paste0(
+    "custom_data/", source,
+    "_level_", level, ".rds"
+  )
 
-  raw_path <- paste0("custom_data/", source, "_level_", level, ".rds")
-  if (!file.exists(raw_path)) {
-    download <- TRUE
-  }
-
-  if (download) {
-    test_that(paste0(data_name, " downloads sucessfully"), {
-      region$download()
-      purrr::walk(region$data$raw, function(data) {
-        expect_s3_class(data, "data.frame")
-        expect_true(nrow(data) > 0)
-        expect_true(ncol(data) >= 2)
-      })
-    })
-    region$data$raw <- purrr::map(region$data$raw,
-      dplyr::slice_tail,
-      n = 250
-    )
-    region$data$raw <- purrr::map(
-      region$data$raw,
-      ~ .[, 1:min(100, ncol(.))]
-    )
-    saveRDS(region$data$raw, raw_path)
-  } else {
-    region$data$raw <- readRDS(raw_path)
-  }
-
-  test_that(paste0(data_name, " can be cleaned as expected"), {
-    region$clean()
-    expect_s3_class(region$data$clean, "data.frame")
-    expect_true(nrow(region$data$clean) > 0)
-    expect_true(ncol(region$data$clean) >= 2)
-    expect_clean_cols(region$data$clean, level = level)
-  })
-
-  test_that(paste0(data_name, " can highlight available regions as expected"), {
-    expect_error(region$available_regions(), NA)
-    expect_true(class(region$available_regions()) %in% "character")
-  })
-
-  test_that(paste0(data_name, " can be processed as expected"), {
-    region$process()
-    expect_s3_class(region$data$processed, "data.frame")
-    expect_true(nrow(region$data$processed) > 0)
-    expect_true(ncol(region$data$processed) >= 2)
-    expect_processed_cols(region$data$processed, level = level)
-    if (!source %in% c("ECDC", "WHO")) {
-      local_region <- region$clone()
-      local_region$localise <- FALSE
-      local_region$process()
-      expect_processed_cols(local_region$data$processed,
-        level = level,
-        localised = FALSE
-      )
-    }
-  })
-
-  test_that(paste0(data_name, " can be returned as expected"), {
-    returned <- region$return()
-    if (any(class(returned) %in% "data.frame")) {
-      expect_s3_class(returned, "data.frame")
-      expect_true(nrow(returned) > 0)
-      expect_true(ncol(returned) >= 2)
-    }
-  })
-
-  expect_columns_contain_data(data_name, region)
-
-  custom_test <- paste0("test_", source, "_level_", level)
-  if (!exists(custom_test)) {
-    custom_test <- function(region) {
-
-    }
-  }
-  do.call(custom_test, list(region = region))
+  # UK nhs data snapshot
+  nhs_snapshot_path <- paste0(
+    "custom_data/", source,
+    "_level_", level, "_nhs", ".rds"
+  )
+  region$test(
+    download = download,
+    snapshot_path = snapshot_path,
+    test_all = TRUE,
+    nhs_included_path = nhs_snapshot_path
```

Comment:
this seems like it will get unmanageable in quite a short time. I think using a folder based path and then controlling the name internally would resolve this (as here fort example you wouldn't need a new path arg per custom data test)

Reply:
there is a world where we might want this as a function in the package as well....